Client/macos/MacOSBluetoothConnector.mm

```diff
@@ -0,0 +1,162 @@
+// #import <Foundation/Foundation.h>
+#include "MacOSBluetoothConnector.h"
+#include <thread>
+
+MacOSBluetoothConnector::MacOSBluetoothConnector()
+{
+	
+}
+MacOSBluetoothConnector::~MacOSBluetoothConnector()
+{
+	//onclose event
+	if (isConnected()){
+		disconnect();
+	}
+}
+
+@interface AsyncCommDelegate : NSObject <IOBluetoothRFCOMMChannelDelegate> {
+    @public
+    MacOSBluetoothConnector* delegateCPP;
+}
+@end
+
+@implementation AsyncCommDelegate {
+}
+// this function fires when the channel is opened
+-(void)rfcommChannelOpenComplete:(IOBluetoothRFCOMMChannel *)rfcommChannel status:(IOReturn)error
+{
+    
+    if ( error != kIOReturnSuccess ) {
+        fprintf(stderr,"Error - could not open the RFCOMM channel. Error code = %08x.\n",error);
+        return;
+    }
+    else{
+        fprintf(stderr,"Connected. Yeah!\n");
+    }
+    
+}
+// this function fires when the channel receives data
+-(void)rfcommChannelData:(IOBluetoothRFCOMMChannel *)rfcommChannel data:(void *)dataPointer length:(size_t)dataLength
+{
+    NSString  *message = [[NSString alloc] initWithBytes:dataPointer length:dataLength encoding:NSUTF8StringEncoding];
+    delegateCPP->dataRec([message UTF8String]);
+}
+
+
+@end
+
+int MacOSBluetoothConnector::send(char* buf, size_t length)
+{
+    fprintf(stderr,"Sending Message\n");
+	//write buffer to channel
+    [(__bridge IOBluetoothRFCOMMChannel*)rfcommchannel writeSync:(void*)buf length:length];
+	return 1; // what to return here?
```

Comment:
number of bytes sent

Reply:
To my knowledge there is no option to return actual number of bytes sent, would it be an option to return the length of the **expected** bytes sent?

Reply:
sure, that should be fine. This program isn't expected to be super robust anyway